web/firefoxcom.js

```diff
@@ -285,6 +285,47 @@ class MozL10n {
   window.addEventListener("editingaction", handleEvent);
 })();
 
+if (PDFJSDev.test("GECKOVIEW")) {
+  (function listenQueryEvents() {
+    window.addEventListener("pdf.js.query", async ({ detail: { queryId } }) => {
+      let result = null;
+      if (queryId === "canDownloadInsteadOfPrint") {
+        result = false;
+        const { pdfDocument, pdfViewer } = PDFViewerApplication;
+        if (pdfDocument) {
+          try {
+            const hasUnchangedAnnotations =
+              pdfDocument.annotationStorage.size === 0;
+            // WillPrint is called just before printing the document and could
+            // lead to have modified annotations.
+            const hasWillPrint =
+              pdfViewer.enableScripting &&
+              !!(await pdfDocument.getJSActions())?.WillPrint;
+            const hasUnchangedOptionalContent = (
+              await pdfDocument.getOptionalContentConfig()
```

Comment:
This will lead to a bunch more allocations/parsing than necessary, please access it via the viewer instead.
Please use `o`
```suggestion
              await pdfViewer.getOptionalContentConfig()
```